pkg/comp-functions/functions/vshnredis/release.go

```diff
@@ -178,22 +194,18 @@ func getReleaseValues(r *v1beta1.Release) (map[string]interface{}, error) {
 }
 
 // setReleaseVersion sets the version from the claim if it's a new instance otherwise it is managed by maintenance function
-func setReleaseVersion(comp *vshnv1.VSHNRedis, values map[string]interface{}) error {
-
-	tag, _, err := unstructured.NestedString(values, "image", "tag")
+func setReleaseVersion(comp *vshnv1.VSHNRedis, values map[string]interface{}, observed map[string]interface{}) error {
+	var err error
+	tag, _, err := unstructured.NestedString(observed, "image", "tag")
 	if err != nil {
 		return fmt.Errorf("cannot get image tag from values in release: %v", err)
 	}
-	// In case the tag is set then let the maintenance manage it
 	if tag != "" {
-		return nil
+		// In case the tag is set, keep the desired
```

Comment:
You mean keep the observed?

Reply:
Yes. You're right I got confused updating these comments :sweat_smile:

Reply:
and I got confused reading the comment :D